PCF-ADR-0003-codeflare-release-process.md

```diff
@@ -0,0 +1,117 @@
+# Standardized release process for the CodeFlare Project
+
+|                |                          |
+| -------------- | ------------------------ |
+| Date           | 04/28/2023               |
+| Scope          |                          |
+| Status         | Proposed                    |
+| Authors        | [Anish Asthana](@anishasthana) [Mustafa Eyceoz](@Maxusmusti) |
+| Supersedes     | N/A                      |
+| Superseded by: | N/A                      |
+| Issues         |                          |
+| Other docs:    | PCF-ADR-002              |
+
+## What
+
+This ADR introduces a unified release process, testing, and versioning scheme for the CodeFlare project.
+
+## Why
+
+The CodeFlare project currently has different release processes and versioning schemes for all of it's subprojects. This creates confusion for users when they wish to use the stack -- what components versions should they be using? How do external dependencies like KubeRay factor in?
+There are also no standards around testing for the project
+This ADR aims to document the overall release process for Project CodeFlare.
+
+## Goals
+
+* Establish a common versioning scheme for all CodeFlare components
+* Establish a release process for the CodeFlare project
+* Establish guidelines for CodeFlare support matrix
+
+## Non-Goals
+
+* Require all components to restart versioning from scratch
+* Outline technical details for tests or CI infrastructure.
+
+## How
+
+We are proposing a unified release cycle for the CodeFlare project.
+We will release a new version of the CodeFlare stack after every sprint (currently every 3 weeks), following normal semantic versioning. All components under the CodeFlare umbrella are expected to start using semantic versioning for future releases. Components are not expected to start versioning again from scratch (i.e v0.0.1). As part of a new release for the CodeFlare Stack, we will update the support matrix which outlines specific component versions and versions for important dependencies such as KubeRay. The components to be documented are as follows:
+
+* CodeFlare Notebook Image
```

Comment:
The notebook image can have releases independent of SDK releases, mainly for urgent fixes/version changes (hence the funky versioning scheme currently), but in an ideal world should usually just release alongside every new SDK release (as it will already). Do we want to consider this as an independent component in the release process, or should that be tied to the SDK as well?

Reply:
I'm good with keeping it as a "part" of the SDK. @MichaelClifford what do you think?

Reply:
Updated.